app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
 
+    people = {}
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
+
+
+def create_person_list(people: list) -> list[Person]:
+    result = [Person(name=data.get("name"),
+                     age=data.get("age")) for data in people]
+
+    for person in people:
+        if wife := person.get("wife"):
+            Person.people[person["name"]].wife = (Person.people[wife])
```

Comment:
```suggestion
            Person.people[person["name"]].wife = Person.people[wife]
```
In this case no need. Because it written in one line

Reply:
The same with husband

Reply:
```suggestion
            Person.people[person["name"]].wife = (
                Person.people[wife]
            )        
```
Like this? if we need new line?